packages/wonder-blocks-core/components/clickable-behavior.js

```diff
@@ -329,15 +329,16 @@ export default class ClickableBehavior extends React.Component<
 
             if (safeWithNav) {
                 if (history && !skipClientNav) {
-                    if (!this.state.waiting) {
-                        // We only show the spinner for safeWithNav when doing
-                        // a server-side navigation since since the spinner is
-                        // indicating that we're waiting for navigation to occur.
-                        this.setState({waiting: true});
-                    }
+                    // client-side nav
                     safeWithNav();
                 } else {
                     try {
+                        if (!this.state.waiting) {
+                            // We only show the spinner for safeWithNav when doing
+                            // a server-side navigation since since the spinner is
```

Comment:
This terminology is a bit confusing (at least for me) as I immediately think of "server-side rendering". Perhaps consider "full page load navigation"?

Reply:
"full page load navigation" is better.  I'll update this before landing.